pypeit/spectrographs/gtc_osiris.py

```diff
@@ -0,0 +1,337 @@
+        par['calibrations']['arcframe']['process']['combine']='weightmean' #Multiple arcs with different lamps, so can't median combine
+        par['calibrations']['standardframe']['exprng'] = [None, 120]
+        par['scienceframe']['exprng'] = [90, None]
+
+        # No ovescan region
+        turn_off = dict(use_overscan=False)
+        par.reset_all_processimages_par(**turn_off)
+        par['scienceframe']['process']['use_overscan'] = False
+
+        par['reduce']['findobj']['find_trim_edge'] = [20,100]
+        par['reduce']['trim_edge'] = [20,100]
+
+
+
+
+        return par
+
+    def init_meta(self):
+        """
+        Define how metadata are derived from the spectrograph files.
+
+        That is, this associates the ``PypeIt``-specific metadata keywords
+        with the instrument-specific header cards using :attr:`meta`.
+        """
+        self.meta = {}
+        # Required (core)
+        self.meta['ra'] = dict(ext=0, card='RA')
+        self.meta['dec'] = dict(ext=0, card='DEC')
+        self.meta['target'] = dict(ext=0, card='object')
+        self.meta['idname'] = dict(ext=0, card='obsmode')
+        self.meta['decker'] = dict(ext=0, card='SLITW')
+        self.meta['binning'] = dict(card=None, compound=True)  # Uses CCDSUM
+        self.meta['detector']=dict(ext=0,card='detector')
+        self.meta['mjd'] = dict(ext=0, card='MJD-OBS')
+        self.meta['exptime'] = dict(ext=0, card='EXPTIME')
+        self.meta['airmass'] = dict(ext=0, card='AIRMASS')
+        self.meta['dispname'] = dict(ext=0, card='GRISM')
+        self.meta['datasec'] = dict(ext=1, card='DATASEC')
+        self.meta['dichroic'] = dict(ext=0, card='FILTER1')
+
+    def compound_meta(self, headarr, meta_key):
+        """
+        Methods to generate metadata requiring interpretation of the header
+        data, instead of simply reading the value of a header card.
+
+        Args:
+            headarr (:obj:`list`):
+                List of `astropy.io.fits.Header`_ objects.
+            meta_key (:obj:`str`):
+                Metadata keyword to construct.
+
+        Returns:
+            object: Metadata value read from the header(s).
+        """
+        if meta_key == 'binning':
+            binspatial, binspec = parse.parse_binning(headarr[0]['CCDSUM'])
+            binning = parse.binning2string(binspec, binspatial)
+            return binning
+
+    def configuration_keys(self):
+        """
+        Return the metadata keys that define a unique instrument
+        configuration.
+
+        This list is used by :class:`~pypeit.metadata.PypeItMetaData` to
+        identify the unique configurations among the list of frames read
+        for a given reduction.
+
+        Returns:
+            :obj:`list`: List of keywords of data pulled from file headers
+            and used to constuct the :class:`~pypeit.metadata.PypeItMetaData`
+            object.
+        """
+        return ['dispname', 'decker', 'binning']
```

Comment:
Can any of the gratings be tilted?  If so, you should include `dispang`

Reply:
No, all setups are fixed.